data-prepper-plugins/sqs-source/src/main/java/org/opensearch/dataprepper/plugins/source/sqs/SqsService.java

```diff
@@ -0,0 +1,126 @@
+ import software.amazon.awssdk.core.client.config.ClientOverrideConfiguration;
+ import software.amazon.awssdk.core.retry.RetryPolicy;
+ import software.amazon.awssdk.services.sqs.SqsClient;
+ import org.opensearch.dataprepper.model.buffer.Buffer;
+ import org.opensearch.dataprepper.model.event.Event;
+ import org.opensearch.dataprepper.model.record.Record;
+ import java.time.Duration;
+ import java.util.ArrayList;
+ import java.util.List;
+ import java.util.concurrent.TimeUnit;
+ import java.util.concurrent.Executors;
+ import java.util.concurrent.ExecutorService;
+ import java.util.stream.Collectors;
+ import java.util.stream.IntStream;
+ 
+ public class SqsService {
+     private static final Logger LOG = LoggerFactory.getLogger(SqsService.class);
+     static final long SHUTDOWN_TIMEOUT = 30L;
+     static final long INITIAL_DELAY = Duration.ofSeconds(20).toMillis();
+     static final long MAXIMUM_DELAY = Duration.ofMinutes(5).toMillis();
+     static final double JITTER_RATE = 0.20;
+    
+     private final SqsSourceConfig sqsSourceConfig;
+     private final SqsEventProcessor sqsEventProcessor;
+     private final SqsClient sqsClient;
+     private final PluginMetrics pluginMetrics;
+     private final AcknowledgementSetManager acknowledgementSetManager;
+     private final List<ExecutorService> allSqsUrlExecutorServices;
+     private final List<SqsWorker> sqsWorkers;
+     private final Buffer<Record<Event>> buffer;
+
+     public SqsService(final Buffer<Record<Event>> buffer,
+                       final AcknowledgementSetManager acknowledgementSetManager,
+                       final SqsSourceConfig sqsSourceConfig,
+                       final SqsEventProcessor sqsEventProcessor,
+                       final PluginMetrics pluginMetrics,
+                       final AwsCredentialsProvider credentialsProvider) {
+                        
+        this.sqsSourceConfig = sqsSourceConfig;
+        this.sqsEventProcessor = sqsEventProcessor;
+        this.pluginMetrics = pluginMetrics;
+        this.acknowledgementSetManager = acknowledgementSetManager;
+        this.allSqsUrlExecutorServices = new ArrayList<>();
+        this.sqsWorkers = new ArrayList<>();
+        this.sqsClient = createSqsClient(credentialsProvider); 
+        this.buffer = buffer;
+     }  
+ 
+
+     public void start() {
+        final Backoff backoff = Backoff.exponential(INITIAL_DELAY, MAXIMUM_DELAY).withJitter(JITTER_RATE)
+                 .withMaxAttempts(Integer.MAX_VALUE);
+                 
+        LOG.info("Starting SqsService");
+
+        sqsSourceConfig.getQueues().forEach(queueConfig -> {
+            String queueUrl = queueConfig.getUrl();
+            int numWorkers = queueConfig.getNumWorkers();
+            ExecutorService executorService = Executors.newFixedThreadPool(
+                    numWorkers, BackgroundThreadFactory.defaultExecutorThreadFactory("sqs-source-new-" + queueUrl));
```

Comment:
Why is there a `-new-` in here?

Also, don't use the queue URL which has special characters. Extract the queue name from the URL and use only that.

```
"sqs-source-" + queueName
```